modules/data.atmosphere/R/download.NOAA.R

```diff
@@ -0,0 +1,387 @@
+        }
+      }
+  }
+  
+  #Update the noaa_data list with the correct data
+  noaa_data[humid_index][[1]] <- humid_data
+  
+  # Convert NOAA's total precipitation (kg m-2) to precipitation flux (kg m-2 s-1)
+  #NOAA precipitation data is an accumulation over 6 hours.
+  precip_index = cf_var_names == "precipitation_flux"
+  
+  noaa_data[precip_index][[1]] = noaa_data[precip_index][[1]] / 21600  #There are 21600 seconds in 6 hours
+  
+  #############################################
+  # Done with data processing.  Now writing the data to the specified directory.
+  if (verbose) {
+    PEcAn.logger::logger.info("Writing data to netcdf file")
+  }
+  
+  
+  if (!dir.exists(outfolder)) {
+    dir.create(outfolder, recursive=TRUE)
+  }
+  
+  setwd(outfolder)
+  
+  #Generate file name.  File names need to account for the specificity in the data.
+  st = lubridate::year(start_date)
+  ed = lubridate::year(end_date)
+  yrcomponent = st
+  if (st != ed) { yrcomponent = paste0(st, "-", ed) } #This could be a function, but it's rather simple and specific for that.
+  
+  st = lubridate::month(start_date)
+  ed = lubridate::month(end_date)
+  moncomponent = st
+  if (st != ed) { moncomponent = paste0(st, "-", ed) }
+  
+  st = lubridate::day(start_date)
+  ed = lubridate::day(end_date)
+  dycomponent = st
+  if (st != ed) { dycomponent = paste0(st, "-", ed) }
+  
+  flname <- if (include_forecast) {
+    paste("NOAA", "GEFS", yrcomponent, moncomponent, dycomponent, "f", "nc", sep=".")
+  } else {
+    paste("NOAA", "GEFS", yrcomponent, moncomponent, dycomponent, "nc", sep=".")
+  }
+  
+  
+  # Create a data frame with information about the file.  This data frame's format is an internal PEcAn standard, and is stored in the BETY database to
+  # locate the data file.  The data file is stored on the local machine where the download occured.
+  results = data.frame(
+    file = file.path(outfolder, flname),  #Path to the file
+    host = PEcAn.remote::fqdn(),          #Name of the server where the file is stored
+    mimetype = "application/x-netcdf",    #Format the data is saved in
+    formatname = "CF Meteorology",        #Type of data
+    startdate = paste0(as.character(start_date), " 00:00:00"),                                     #starting date and time, down to the second
+    enddate = paste0(as.character(end_date), " ", substring(last_data_set_at, 1, 2), ":00:00"),    #ending date and time, down to the second
+    dbfile.name = "NOAA.GEFS",            #Source of data
+    stringsAsFactors = FALSE
+  )
+  
+  #Begin working with the netcdf4 saving format
+  #define dimensions.  Note that all variables will have the same dimension, so we can reuse them for all variables.  The data is two dimensional: one dimension
+  #is time (measured in periods of six hours), while the other one is which particular ensemble the data is drawn from.
```

Comment:
1) PEcAn standard requires lat and lon dimensions, even if they are length 1.
2) I believe our current standard is to write different ensemble members to different files. @ankurdesai @crollinson can you confirm and point @Luke-Dramko to an example of where/how you're writing out met ensembles in the downscaling code, as well as how you're storing ensembles in the database (i.e. is the input & dbfiles ID by site or by ensemble member?)

Reply:
Currently, I use a hierarchical naming scheme where it's ID.ENS (adding on another .ENS if you're doing both bias correction & temporal downscaling) and each member gets its own folder.  

In my workflow, I specify the [ensemble member tags up front](https://github.com/crollinson/MetDownscaling_Manuscript/blob/505a6a602aeb686618e78943c420e585192227c1/Workflow_Downscale/3_bias_correct.R#L57-L59) (allowing me to easily add on), which then get passed to the downscaling argument ([bias correction example](https://github.com/crollinson/MetDownscaling_Manuscript/blob/505a6a602aeb686618e78943c420e585192227c1/Workflow_Downscale/3_bias_correct.R#L108))

I don't know how this is getting implemented in PEcAn's met workflow.  I did try to make sure it would be compatible with what I know of PEcAn's arrangements, but also make it flexible enough for my standalone use.  Hope this helps!